cmd/create/machinepool/cmd.go

```diff
@@ -400,6 +419,51 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	isSpotSet := cmd.Flags().Changed("use-spot-instances")
+	isSpotMaxPriceSet := cmd.Flags().Changed("spot-max-price")
+
+	useSpotInstances := args.useSpotInstances
+	spotMaxPrice := args.spotMaxPrice
+	if isSpotMaxPriceSet && isSpotSet && !useSpotInstances {
+		reporter.Errorf("Can't set max price when not using spot instances")
+		os.Exit(1)
+	}
+
+	if isSpotMaxPriceSet {
+		isSpotSet = true
+		useSpotInstances = true
+	}
+
+	if !isSpotSet && interactive.Enabled() {
+		useSpotInstances, err = interactive.GetBool(interactive.Input{
+			Question: "Use spot instances",
+			Help:     cmd.Flags().Lookup("use-spot-instances").Usage,
+			Default:  useSpotInstances,
+			Required: false,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid value for use spot instances: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	if useSpotInstances && !isSpotMaxPriceSet && interactive.Enabled() {
+		spotMaxPrice, err = interactive.GetFloat(interactive.Input{
+			Question: "Spot instance max price",
+			Help:     cmd.Flags().Lookup("spot-max-price").Usage,
+			Required: false,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid value for spot max price: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	if spotMaxPrice < 0 {
+		reporter.Errorf("Spot max price must be positive")
+		os.Exit(1)
+	}
```

Comment:
@vkareh @zgalor spot instances are not yet supported in OSD. I assume this is a capability that can be enabled for orgs?  Can we make these flags hidden?

Reply:
@jharrington22 they are supported for AWS CCS clusters - which is why I only added them to rosa and not ocm cli

Reply:
cc @igoihman

Reply:
We already merged this in OCM backend for CCS/AWS without capability gating